schema.json

```diff
@@ -1344,6 +1375,183 @@
         "type": "color",
         "label": "Text hover color",
         "id": "card--alternate-color--hover"
+      },
+      {
+        "type": "heading",
+        "content": "Image sizes"
+      },
+      {
+        "type": "paragraph",
+        "content": "Please specify a maximum image size for desktop display. We'll automatically scale down the image for smaller devices.",
+        "hideExpression": "model.productSize != 'product'"
+      },
+      {
+        "type": "select",
+        "label": "Main product images",
+        "id": "productSize",
+        "force_reload": true,
+        "options": [
+          {
+            "value": "500x659",
+            "label": "Optimized for theme"
+          },
+          {
+            "value": "product",
+            "label": "Specify dimensions"
+          }
+        ]
+      },
+      {
+        "type": "number",
+        "id": "_images.product.width",
+        "force_reload": true,
+        "label": "Max width",
+        "className": "formly-field-dimensions",

```

Comment:
ideally, I would like width x height to be one type, this is definitely hacky, I can move the classNames to theme-editor for now, but it's just as hacky, we should definitely do a tech debt on this, but fyi this needs be in asap.